src/Symfony/Component/RateLimiter/LimitInterface.php

```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Symfony\Component\RateLimiter;
+
+interface LimitInterface
+{
+    public function getRetryAfter(): \DateTimeImmutable;
+    public function getRemainingTokens(): int;
+}
```

Comment:
I think it would be good to add a third method: `isAccepted()` (or the like). This returns true/false if the limiter accepts or blocks this hit.

Or we have to decide to throw a specific exception if the limit blocks, something suggested by @kbond in the issue.

Reply:
What about:

1. `getStrategy(): string` for the strategy name (`fixed_window`, `token_bucket`)
2. `getMetadata(): array` for any other strategy specific details?

Reply:
getStrategy is a bit ugly to pass but doable :)

Reply:
You can directly pass the strings from the limiters - each limiter class is a separate strategy